index.js

```diff
@@ -3,6 +3,7 @@
 const Client = require('./lib/core/client')
 const errors = require('./lib/core/errors')
 const Pool = require('./lib/pool')
+const { Agent, request, stream, pipeline, setGlobalAgent } = require('./lib/agent')
```

Comment:
@ronag so this is causing a circular dependency issue. 

`lib/agent.js` requires `Pool` from this file since it needs the prototype methods on lines 14-18. How would you like me to resolve this? Can I move the prototype changes to `lib/pool.js`?

Reply:
What prototype methods? I don't see how this is circular?

Reply:
lines 14-18 are like `Pool.prototype.request = require('client-request.js')`

Agent requires access to this modded Pool class rather than the one exported from `./lib/pool`. 

So since `agent.js` requires `../index.js` and `index.js` requires `./lib/agent` its a circular ref

Reply:
Implement pool in agent with require('./lib/client-request') etc...

Reply:
alterantively move the pool modding to the pool file